eips_code/eof1_validation.py

```diff
@@ -0,0 +1,46 @@
+from eip4750 import * 
+from eip5450 import validate_function, FunctionType
+
+
+# EOF code validation, including container format validation, code section validation and function stack validation.
+# Raises ValidationException on invalid code
+def validate_eof1(container: bytes):
+    validate_eof(container)
+
+    section_sizes, header_size = read_eof1_header(container)
+
+    if len(section_sizes[S_TYPE]) != 0:
+        type_section_size = section_sizes[S_TYPE][0]
+        types = read_eof1_types(container, header_size, type_section_size)
+        pos = header_size + type_section_size
+    else:
+        types = [FunctionType(0, 0)] # implicit code section 0 type
+        pos = header_size
+
+    for func_idx in range(0, len(section_sizes[S_CODE])):
+        code_section_size = section_sizes[S_CODE][func_idx]
+        validate_function(func_idx, container[pos:pos + code_section_size], types)
+        pos += code_section_size
+
+def read_eof1_header(container: bytes) -> (dict[int, list[int]], int):
```

Comment:
Maybe just define a class? See e.g. https://github.com/ipsilon/eof/blob/stack_validation/eips_code/eof1.py#L9-L12.

Reply:
Is it better now? Got somewhat more complicated maybe

Reply:
Now it's better.